doc/architecture/decisions/0015-container-api.md

```diff
@@ -0,0 +1,103 @@
+# 15. Container API
+
+Date: 2018-09-17
+
+## Status
+
+Proposed
+
+## Context
+
+* We have several different roles for using a Holochain app (through a *container*):
```

Comment:
Has the department of names department come up with a better name than "container" that isn't so overloaded with VM implications yet? Suggest we sort that out before merging this.

Reply:
@lucksus can you explain on a practical level what a "container" is?

Reply:
@lucksus is a "container" a chrooted user? is it somehow tied to linux machine users? etc trying to understand how the operating system will know about the "boundaries" if this container

Reply:
@lucksus answered in the call that "container" == a library

Reply:
I actually like your distinctions from [your HDK doc](https://docs.google.com/document/d/1lFRQxrJqGI97A1S6WGsoL84qlVkP7HVjVF4cVA7zUF8/edit#) of Holochain System (for Container) and Holochain Service (for process that uses the System.

I mean system and service are kind of general words, but they seem to work.